src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3677,36 +3676,46 @@ size_t os::Linux::find_large_page_size(size_t large_page_size) {
     if (entry->d_type == DT_DIR &&
         sscanf(entry->d_name, "hugepages-%zukB", &page_size) == 1) {
       // The kernel is using kB, hotspot uses bytes
-      if (large_page_size == page_size * K) {
-        closedir(dir);
-        return large_page_size;
+      if (page_size * K > (size_t)Linux::page_size()) {
+          // Add each found Large Page Size to _page_sizes
+          _page_sizes.add(page_size * K);
       }
     }
   }
   closedir(dir);
-  return _default_large_page_size;
+
+  LogTarget(Info, pagesize) lt;
+  if (lt.is_enabled()) {
+    LogStream ls(lt);
+    ls.print("Available page sizes: ");
+    _page_sizes.print_on(&ls);
+  }
```

Comment:
Does this work and show something? I know UL is initialization time sensitive (which is annoying btw).

Reply:
This comes from I comment I made and UL is initialized here. Not sure this is exactly where this should end up since it will only be printed when large pages are enabled. I think it might make sense to move somewhere else or make it a completely separate change.